### PR TITLE
Loadout filter pills

### PR DESCRIPTION
**Pull request by bhollis**
Screenshot 2022-12-24 at 5 12 32 PM
Screenshot 2022-12-24 at 5 11 45 PM

This adds what I think is my preferred solution to #7663 - autogenerated filter buttons based on hashtags used in the name or notes for loadouts. These show up automatically when you use hashtags, and can be clicked to quickly filter down to a specific set of loadouts. Shift-clicking ANDs them together to drill down even further. I also added some automatic filters for missing items and deprecated mods to help with cleanup.

### CODE REVIEW

**Comment by bhollis** [comment]
I'll probably feature-flag these before merging

**Comment by nev-r** [comment]
this is really, really cool.

File: src/app/loadout/loadout-ui/menu-hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  const filterOptions = Object.keys(loadoutsByHashtag).map(
+    (hashtag): Option => ({
+      key: hashtag,
+      content: hashtag,
+    })
+  );
+
+  let loadoutsWithMissingItems: Loadout[] = [];
+  let loadoutsWithDeprecatedMods: Loadout[] = [];
+  loadoutsWithMissingItems = useMemo(
```
Comment (robojumper) [inline src/app/loadout/loadout-ui/menu-hooks.tsx:71]: Doesn't look like these are reassigned later.

**Review by robojumper — COMMENTED** [review]
I like it a lot. Two small things I've noticed:

grafik
 i.e. filtering with shift-clicking on pills so that there are no matching loadouts blames an empty query string.

**Pull request merged**